In [111]:
# Basic Libraries
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt # we only need pyplot
sb.set() # set the default Seaborn style for graphics

In [112]:
DrugData = pd.read_csv('Drug.csv')
DrugData.head()


Condition          Drug Indication Type      Reviews  \
0  Acute Bacterial Sinusitis  Levofloxacin   On Label   RX  994 Reviews   
1  Acute Bacterial Sinusitis  Levofloxacin   On Label   RX  994 Reviews   
2  Acute Bacterial Sinusitis  Moxifloxacin   On Label   RX  755 Reviews   
3  Acute Bacterial Sinusitis  Azithromycin   On Label   RX  584 Reviews   
4  Acute Bacterial Sinusitis  Azithromycin   On Label   RX  584 Reviews   

   Effective  EaseOfUse  Satisfaction  \
0       2.52       3.01          1.84   
1       2.52       3.01          1.84   
2       2.78       3.00          2.08   
3       3.21       4.01          2.57   
4       3.21       4.01          2.57   

                                         Information  
0  \n\t\t\t\t\tLevofloxacin is used to treat a va...  
1  \n\t\t\t\t\tLevofloxacin is used to treat a va...  
2  \n\t\t\t\t\t This is a generic drug. The avera...  
3  \n\t\t\t\t\tAzithromycin is an antibiotic (mac...  
4  \n\t\t\t\t\tAzithromycin is an antibiotic (mac...

In [113]:
DrugData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2219 entries, 0 to 2218
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Condition     2219 non-null   object 
 1   Drug          2219 non-null   object 
 2   Indication    2219 non-null   object 
 3   Type          2219 non-null   object 
 4   Reviews       2219 non-null   object 
 5   Effective     2219 non-null   float64
 6   EaseOfUse     2219 non-null   float64
 7   Satisfaction  2219 non-null   float64
 8   Information   2219 non-null   object 
dtypes: float64(3), object(6)
memory usage: 156.1+ KB


Let's list down all our observations from the super-quick glance of the dataset, as above.

There are 9 variables/features/columns and 2219 observations/samples/rows in the dataset.

The response variable seems to be Effectiveness, while the remaining 8 are most likely predictors.

There are 3 variables identified as float64 by default, and it seems they are indeed Numeric.

There are 6 variables identified as object by default, and they are most likely Categorical (except Reviews,which seems to be numerical count).

None of the variables/features seem to have any missing value.

---
## Data Preparation and Cleaning

We will look through the dataset to prepare and indentify any areas that needs to be cleaned before our analysis.

In [114]:
DrugData["Reviews"]

0       994 Reviews
1       994 Reviews
2       755 Reviews
3       584 Reviews
4       584 Reviews
           ...     
2214      2 Reviews
2215      1 Reviews
2216      1 Reviews
2217      1 Reviews
2218      1 Reviews
Name: Reviews, Length: 2219, dtype: object

Looking at the Reviews Variable, we note that Reviews should be Numerical but it is classified as object64 due to the string behind. Hence, there is a need to remove "reviews" and convert it to a numerical variable.

In [115]:

DrugData["Reviews"] = DrugData["Reviews"].map(lambda x: x.rstrip(' Reviews')).astype('int64')
DrugData["Reviews"]


0       994
1       994
2       755
3       584
4       584
       ... 
2214      2
2215      1
2216      1
2217      1
2218      1
Name: Reviews, Length: 2219, dtype: int64

The price and form of medicine can be found within the information column. We will then extract the price and form from the column and create new columns for price and form. (Example of forms are "tablet", "capsule" etc.)

In [119]:
all_form = ['Capsule(s)','Capsule','Tablet(s)','Tablet','tablets','Bottle','Vial(s)','Vial','Reconstituted(s)','Reconstituted','Tube','Jar','Can','Box','Syringe','Implant','Package','Pen(s)','Inhaler']
Form = []
Price = []
for i in range (DrugData.shape[0]):
    info = DrugData.iloc[i,8]
    sentence = info.split('.')
    for x in sentence:
        if 'average' in x :
            words = x.split()
    for word in words:
        word = word.replace(",", "")
        if word in all_form :
            Form.append(word)
            break
    for price in words:
        if '$' in price:
            temp = price.replace("$", "")
            Price.append(temp)

172
17
172
17
32
3
32
3
32
3
32
3
32
3
32
3
32
3
32
3
172
17
172
17
12
1
12
1
12
1
12
1
99
9
99
9
99
9
99
9
99
9
99
9
99
9
125
12
125
12
287
28
99
9
172
17
32
3
125
12
125
12
161
16
332
33
332
33
12
1
12
1
12
1
12
1
12
1
12
1
161
16
332
33
99
9
332
33
332
33
125
12
125
12
12
1
12
1
12
1
161
16
161
16
12
1
12
1
12
1
12
1
12
1
332
33
332
33
332
33
332
33
5829
582
362
36
2049
204
10362
1036
170
17
170
17
5829
582
1516
151
23
2
23
2
2049
204
23
2
362
36
10362
1036
11
1
11
1
11
1
33
3
15
1
10
1
33
3
33
3
11
1
11
1
16
1
15
1
15
1
15
1
11
1
11
1
33
3
11
1
10
1
33
3
11
1
302
30
302
30
96
9
16
1
156
15
53
5
206
20
7323
732
7323
732
53
5
206
20
15
1
15
1
74
7
206
20
74
7
3099
309
11
1
10
1
13
1
13
1
15
1
11
1
11
1
206
20
14
1
206
20
206
20
49
4
11
1
16
1
15
1
11
1
16
1
15
1
10
1
33
3
10
1
49
4
11
1
11
1
16
1
49
4
16
1
16
1
16
1
11
1
16
1
16
1
11
1
10
1
10
1
11
1
11
1
14
1
348
34
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
11
1
7

11
1
11
1


In [117]:
tablet = ['Tablet(s)','Tablet','tablets']
capsule = ['Capsule(s)','Capsule']
cream = ['Tube','Can','Jar']
liquid_drink = ['Bottle']
liquid_inject = ['Vial','Reconstituted','Reconstituted(s)','Vial(s)','Pen(s)','Syringe']
other = ['Box','Package','Implant','Inhaler']
for i in range (len(Form)):
    if Form[i] in tablet:
        Form[i] = 'Tablet'
    elif Form[i] in capsule:
        Form[i] = 'Capsule'
    elif Form[i] in cream:
        Form[i] = 'Cream'
    elif Form[i] in liquid_drink :
        Form[i] = 'Liquid (Drink)'
    elif Form[i] in liquid_inject :
        Form[i] = 'Liquid (Inject)'
    elif Form[i] in other:
        Form[i] = 'Other'